File: python/logger.py


class Logger( object ):
    def __init__( self, prefix ):
        self.prefix = prefix

    def write( self, lvl, text ):
        print "[%s] %s - %s" % (lvl, self.prefix, text)


    def info( self, text ):
        self.write( "INFO", text )


    def warning( self, text ):
        self.write( "WARNING", text )


    def error( self, text ):
        self.write( "ERROR", text )


    def debug( self, text ):
        self.write( "DEBUG", text )



def init_logging():
    pass


def getLogger( prefix ):
    return Logger( prefix )

File: java/src/main/java/addresslabel/util/Logger.java
package addresslabel.util;

public class Logger
{
    private String _prefix;

    public Logger( String prefix )
    {
        _prefix = prefix;
    }

    public static Logger getLogger( String prefix )
    {
        return new Logger( prefix );
    }

    public void write( String lvl, String text )
    {
        System.out.println( String.format( "[%s] %s - %s", lvl, _prefix, text ) );
    }


    public void info( String text )
    {
        write( "INFO", text );
    }


    public void warning( String text )
    {
        write( "WARNING", text );
    }


    public void error( String text )
    {
        write( "ERROR", text );
    }


    public void debug( String text )
    {
        write( "DEBUG", text );
    }
}




